helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -207,6 +207,16 @@ public Builder(CloudConfig cloudConfig) {
       _cloudInfoProcessorName = cloudConfig.getCloudInfoProcessorName();
     }
 
+    public Builder(String cluster, boolean enabled, CloudProvider cloudProvider, String cloudID,
```

Comment:
Why we need this constructor while all the fields can be specified with the specific methods?

Reply:
I just added this constructor for simplicity. It is not needed but if we can avoid calling several set function and use this function instead, the code will be more readable. If you have strong preference about it, I can remove it though.